### PR TITLE
Created delete wallet graphql service

### PR DESCRIPTION
**Pull request by johnwu93**
Created unit tests to test the behavior of the service.

Here is proof of the service working:

# Success Case
1. `mutation{addWallet{publicKey}}`

```graphql
{
  "data": {
    "addWallet": {
      "publicKey": "ASh7a/qM8l2jynPF/zTgF2BbiR5tJvip7GMMawQAs+M6r+btUmZUAAAAAA=="
    }
  }
}
```

2. `{ownedWallets{publicKey, privateKeyPath}}`

```graphl
{
  "data": {
    "ownedWallets": [
      {
        "publicKey": "ASh7a/qM8l2jynPF/zTgF2BbiR5tJvip7GMMawQAs+M6r+btUmZUAAAAAA==",
        "privateKeyPath": "/Users/johnwu/.coda-config/wallets/store/ASh7axqM8l2jynPFxzTgF2BbiR5tJvip7GMMawQAs+M6r+btUmZUAAAAAA=="
      }
    ]
  }
}
```

3. 
```graphql
mutation{deleteWallet (input: 
  	{publicKey: "ASh7a/qM8l2jynPF/zTgF2BbiR5tJvip7GMMawQAs+M6r+btUmZUAAAAAA=="} ) {
    publicKey
  }}
```

```
{
  "data": {
    "deleteWallet": {
      "publicKey": "ASh7axqM8l2jynPFxzTgF2BbiR5tJvip7GMMawQAs+M6r+btUmZUAAAAAA==""
    }
  }
}
```

4. `{ownedWallets{publicKey, privateKeyPath}}`

```
{
  "data": {
    "ownedWallets": []
  }
}
```

# Failed Case
There are no existing wallets

```
mutation{deleteWallet (input: 
  	{publicKey: "ASh7a/qM8l2jynPF/zTgF2BbiR5tJvip7GMMawQAs+M6r+btUmZUAAAAAA=="} ) {
    publicKey
  }}
```

```
{
  "errors": [
    {
      "message": "Could not find wallet with public key: ASh7a/qM8l2jynPF/zTgF2BbiR5tJvip7GMMawQAs+M6r+btUmZUAAAAAA==",
      "path": [
        "deleteWallet"
      ]
    }
  ],
  "data": null
}
```
Checklist:

- [ ] Tests were added for the new behavior
- [X] All tests pass (CI will check this if you didn't)
- [ ] Serialized types are in stable-versioned modules
- [ ] Does this close issues? List them:

### CODE REVIEW

File: src/app/cli/src/graphql.ml [hunk 1]
```diff
@@ -431,6 +435,10 @@ module Make (Commands : Coda_commands.Intf) = struct
             ; fee ~doc:"Fee amount in order to send a stake delegation"
             ; memo ~doc:"Public description of a stake delegation" ]
 
+      let delete_wallet =
+        obj "deleteWallet" ~coerce:Fn.id
+          ~fields:[arg "publicKey" ~typ:(non_null string)]
```
Comment (Schmavery) [inline src/app/cli/src/graphql.ml]: It looks like you defined this twice?

File: src/lib/secrets/wallets.ml [hunk 2]
```diff
@@ -58,6 +58,12 @@ let generate_new t : Public_key.Compressed.t Deferred.t =
   Public_key.Compressed.Table.add_exn t.cache ~key:pk ~data:keypair ;
   pk
 
+let delete ({cache; _} as t : t) (pk : Public_key.Compressed.t) :
+    (unit, [`Not_found]) Deferred.Result.t =
+  Hashtbl.remove cache pk ;
+  Deferred.Or_error.try_with (fun () -> Unix.remove (get_path t pk))
+  |> Deferred.Result.map_error ~f:(fun _ -> `Not_found)
```
Comment (Schmavery) [inline src/lib/secrets/wallets.ml]: Seems like conceivably there could be more errors than Not_found? This kinda conflates Not_found in the list of wallets and Not_found on disk/some error deleting. One could argue that not_found on disk actually a success..
Maybe we don't need to differentiate them, not sure

Comment (johnwu93) [reply src/lib/secrets/wallets.ml]: I don't think we need to differentiate at the time being

File: src/app/cli/src/graphql.ml (hunk 1 above)
Comment (johnwu93) [reply src/app/cli/src/graphql.ml]: oh no, one of them is defined in inputs and one of them is defined in payload

**Pull request merged**